siliconcompiler/schema/schema_cfg.py

```diff
@@ -1232,62 +1234,60 @@ def schema_datasheet(cfg, name='default', mode='default'):
-            and per pin basis.""")
+                "cli: -datasheet_package_pin_shape 'abcd B1 circle'",
+                "api: chip.set('datasheet', 'package', 'abcd', 'pin', 'B1', 'shape', 'circle')"],
+            schelp="""Datasheet: package pin shape (rectangle or circle) specified on a per package
+            and per pin number basis.""")
 
-    metrics = {'pinwidth': ['pinwidth', (0.2, 0.25, 0.3), 'mm'],
-               'pinlength': ['pinlength', (0.2, 0.25, 0.3), 'mm']
+    metrics = {'width': ['width', (0.2, 0.25, 0.3), 'mm'],
+               'length': ['length', (0.2, 0.25, 0.3), 'mm']
                }
 
     for i, v in metrics.items():
-        scparam(cfg, ['datasheet', 'package', name, i, name],
+        scparam(cfg, ['datasheet', 'package', name, 'pin', pinnumber, i],
                 unit=v[2],
                 sctype='(float,float,float)',
                 shorthelp=f"Datasheet: package pin {v[0]}",
-                switch=f"-datasheet_package_{i} 'name name <(float,float,float)>'",
+                switch=f"-datasheet_package_pin_{i} 'name pinnumber <(float,float,float)>'",
                 example=[
-                    f"cli: -datasheet_package_{i} 'abcd B1 {v[1]}'",
-                    f"api: chip.set('datasheet', 'package', 'abcd', '{i}', 'B1', {v[1]}"],
-                schelp=f"""Datsheet: {v[0]} specified on a per package and per pin basis.
-                Values are tuples of (min, nominal, max).""")
+                    f"cli: -datasheet_package_pin_{i} 'abcd B1 {v[1]}'",
+                    f"api: chip.set('datasheet', 'package', 'abcd', 'pin', 'B1', '{i}', {v[1]}"],
+                schelp=f"""Datsheet: {v[0]} specified on a per package and per pin number
+                basis. Values are tuples of (min, nominal, max).""")
 
-    scparam(cfg, ['datasheet', 'package', name, 'pinloc', name],
+    scparam(cfg, ['datasheet', 'package', name, 'pin', pinnumber, 'loc'],
             sctype='(float,float)',
             unit='mm',
             shorthelp="Datasheet: package pin location",
-            switch="-datasheet_package_pinloc 'name name <(float,float)>'",
+            switch="-datasheet_package_pin_loc 'name pinnumber <(float,float)>'",
             example=[
-                "cli: -datasheet_package_pinloc 'abcd B1 (0.5,0.5)'",
-                "api: chip.set('datasheet', 'package', 'abcd', 'pinloc', 'B1', (0.5,0.5)"],
-            schelp="""Datsheet: Pin location specified as an (x,y) tuple. Locations
-            specify the center of the pin with respect to the center of the package.
-            """)
+                "cli: -datasheet_package_pin_loc 'abcd B1 (0.5,0.5)'",
+                "api: chip.set('datasheet', 'package', 'abcd', 'pin', 'B1', 'loc', (0.5,0.5)"],
+            schelp="""Datsheet: Package pin location specified as an (x,y) tuple on a per
+            package and per pin number basis. Locations specify the center of the pin with
+            respect to the center of the package.""")
```

Comment:
I'm not sure about the center of the package thing, it feels awkward to define things relative to the center at the package level

Reply:
Package pins and any interactions withh the OSATs always refers to the origin being at the center of the package. In the chip world, the origin is in the lower left.